core/trino-main/src/main/java/io/trino/execution/scheduler/SubnetBasedTopology.java

```diff
@@ -100,7 +100,7 @@ private static void validateHierarchy(List<Integer> lengths, AddressProtocol pro
         }
 
         if (!lengths.isEmpty()) {
-            if (lengths.get(0) <= 0 || Iterables.getLast(lengths) >= protocol.getTotalBitCount()) {
+            if (lengths.get(0) <= 0 || findLast(lengths.stream()).get() >= protocol.getTotalBitCount()) {
```

Comment:
this cases a false positive warning for me in my IDE. i shouldn't need to ignore warnings with my eyes, as this makes me less likely to nice those that matter

use `.orElseThrow()` at least but actually `Iterables.getLast` should be preferred, because this is the **simplest way to convey the intent**.

btw `Iterables.getLast` is O(1) for random-access lists, and `Streams.findLast` is... idk. _Maybe_ O(1) too.

Reply:
Javodoc suggests it will not be O(1):
```
This method's runtime will be between O(log n) and O(n), performing better on efficiently splittable  streams.
```

Reply:
Right. I got fooled by its attempt to split spliterators the right way, but it indeed cannot do better than O(log n).

Reply:
> this cases a false positive warning for me in my IDE. i shouldn't need to ignore warnings with my eyes, as this makes me less likely to nice those that matter

I was thinking if this check should be disabled in IDE. It find it rather annoying to try to come up with a clever exception / error message every time it is needed to dereference an Optional that is never expected to be null. 

> btw Iterables.getLast is O(1) for random-access lists, and Streams.findLast is... idk. Maybe O(1) too.

Nice catch. Let me exclude this for now.

Reply:
>  was thinking if this check should be disabled in IDE. 

we can discuss this separately at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#additional-ide-configuration .
this PR isn't really specific around `Optional.get` usage.

Reply:
Updated